### PR TITLE
Refactor clear coat to be more usable in game

### PR DESCRIPTION
**Pull request by sebastienlagarde**
This PR refactor the clear coat to have correct performance and correct visual with simpler code.

It doesn't perform any refract for direct lighting or area light (not visible anyway) and rely mainly on the IBL coating. 
The clear coat modify the roughness and the fresnel0 of the base layer + try to do very approximate energy conservation (over simplified for performance)

The clear coat have fixed roughness and F0 value and use a coatMask only as parameter.

+ rename  preLightData.NdotV to  preLightData.clampNdotV;

### CODE REVIEW

File: ScriptableRenderPipeline/Core/CoreRP/ShaderLibrary/CommonMaterial.hlsl [hunk 1]
```diff
@@ -45,11 +41,64 @@ void ConvertAnisotropyToRoughness(real perceptualRoughness, real anisotropy, out
...
+}
+
+// This function is a coarse approximation of computing fresnel0 for a different top than air (here clear coat of IOR 1.5) when we only have fresnel0 with air interface
+// This function is equivalent to IORToFresnel0(Fresnel0ToIor(fresnel0), 1.5)
+// mean
+// real sqrtF0 = sqrt(fresnel0);
+// return Sq(1.0 - 5.0 * sqrtF0) / Sq(5.0 - sqrtF0);
+// Optimization: Fit of the function (3 mad) for range 0.04 (should return 0), 1 (should return 1)
+// return saturate(-0.0256868 + fresnel0 * (0.326846 + (0.978946 - 0.283835 * fresnel0) * fresnel0));
+TEMPLATE_1_REAL(Fresnel0ReajustFor15, fresnel0, return saturate(-0.0256868 + fresnel0 * (0.326846 + (0.978946 - 0.283835 * fresnel0) * fresnel0)) )
```
Comment (EvgeniiG) [inline ScriptableRenderPipeline/Core/CoreRP/ShaderLibrary/CommonMaterial.hlsl]: Typo, should be "Readjust"? Strange name, anyway. Better use something like "Approx".

File: ScriptableRenderPipeline/Core/CoreRP/ShaderLibrary/CommonMaterial.hlsl [hunk 2]
```diff
@@ -45,11 +41,64 @@ void ConvertAnisotropyToRoughness(real perceptualRoughness, real anisotropy, out
...
+}
+
+// ior is a value between 1.0 and 2.5
+// Assume air interface for top
+real IORToFresnel0(real ior)
+{
+    return Sq((ior - 1.0) / (ior + 1.0));
+}
+
+real IORToFresnel0(real baseIor, real topIor)
```
Comment (EvgeniiG) [inline ScriptableRenderPipeline/Core/CoreRP/ShaderLibrary/CommonMaterial.hlsl]: Inconsistent capitalization with IORToFresnel0.

File: ScriptableRenderPipeline/Core/CoreRP/ShaderLibrary/CommonMaterial.hlsl [hunk 3]
```diff
@@ -45,11 +41,64 @@ void ConvertAnisotropyToRoughness(real perceptualRoughness, real anisotropy, out
...
     roughnessB = ClampRoughnessForAnalyticalLights(roughnessB);
 }
 
+// Use with stack BRDF (clear coat / coat)
+real roughnessToVariance(real roughness)
+{
+    return 2.0 / Sq(roughness) - 2.0;
+}
+
+real varianceToRoughness(real variance)
```
Comment (EvgeniiG) [inline ScriptableRenderPipeline/Core/CoreRP/ShaderLibrary/CommonMaterial.hlsl]: Should be capitalized.

File: ScriptableRenderPipeline/Core/CoreRP/ShaderLibrary/CommonMaterial.hlsl [hunk 4]
```diff
@@ -45,11 +41,64 @@ void ConvertAnisotropyToRoughness(real perceptualRoughness, real anisotropy, out
...
+void ConvertAnisotropyToClampRoughness(real perceptualRoughness, real anisotropy, out real roughnessT, out real roughnessB)
+{
+    ConvertAnisotropyToRoughness(perceptualRoughness, anisotropy, roughnessT, roughnessB);
 
     roughnessT = ClampRoughnessForAnalyticalLights(roughnessT);
     roughnessB = ClampRoughnessForAnalyticalLights(roughnessB);
 }
 
+// Use with stack BRDF (clear coat / coat)
+real roughnessToVariance(real roughness)
```
Comment (EvgeniiG) [inline ScriptableRenderPipeline/Core/CoreRP/ShaderLibrary/CommonMaterial.hlsl]: Should be capitalized.

File: ScriptableRenderPipeline/Core/CoreRP/ShaderLibrary/CommonMaterial.hlsl [hunk 5]
```diff
@@ -45,11 +41,64 @@ void ConvertAnisotropyToRoughness(real perceptualRoughness, real anisotropy, out
...
+// This function is equivalent to IORToFresnel0(Fresnel0ToIor(fresnel0), 1.5)
+// mean
+// real sqrtF0 = sqrt(fresnel0);
+// return Sq(1.0 - 5.0 * sqrtF0) / Sq(5.0 - sqrtF0);
+// Optimization: Fit of the function (3 mad) for range 0.04 (should return 0), 1 (should return 1)
+// return saturate(-0.0256868 + fresnel0 * (0.326846 + (0.978946 - 0.283835 * fresnel0) * fresnel0));
+TEMPLATE_1_REAL(Fresnel0ReajustFor15, fresnel0, return saturate(-0.0256868 + fresnel0 * (0.326846 + (0.978946 - 0.283835 * fresnel0) * fresnel0)) )
+
+// same as regular refract except there is not the test for total internal reflection + the vector is flipped for processing
+real3 CoatRefract(real3 X, real3 N, real ieta)
```
Comment (EvgeniiG) [inline ScriptableRenderPipeline/Core/CoreRP/ShaderLibrary/CommonMaterial.hlsl]: What happens in the case of the total internal refraction?

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/Editor/Material/Lit/LitUI.cs [hunk 6]
```diff
@@ -58,8 +58,7 @@ protected static class Styles
             public static GUIContent thicknessRemapText = new GUIContent("Thickness Remap", "Remaps values of the thickness map from [0, 1] to the specified range.");
 
             // Clear Coat
-            public static GUIContent coatCoverageText = new GUIContent("Coat Coverage", "Percentage of clear coat coverage");
-            public static GUIContent coatIORText = new GUIContent("Coat IOR", "IOR of clear coat, value is [0..1] + 1.0. i.e 0.5 is IOR 1.5");
+            public static GUIContent coatMaskText = new GUIContent("Coat Mask", "attenuate the coating effect (similar to change to IOR of 1");
```
Comment (EvgeniiG) [inline ScriptableRenderPipeline/HDRenderPipeline/HDRP/Editor/Material/Lit/LitUI.cs]: Capitalization

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl [hunk 7]
```diff
@@ -71,6 +71,13 @@ TEXTURE2D_ARRAY(_LtcData); // We pack the 3 Ltc data inside a texture array
 #define LTC_LUT_SCALE  ((LTC_LUT_SIZE - 1) * rcp(LTC_LUT_SIZE))
 #define LTC_LUT_OFFSET (0.5 * rcp(LTC_LUT_SIZE))
 
+// Constant value for clear coat
+#define CLEAR_COAT_IOR 1.5
+#define CLEAR_COAT_IETA (1.0 / CLEAR_COAT_IOR)
```
Comment (EvgeniiG) [inline ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl]: What is IETA?

Comment (EvgeniiG) [reply ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl]: Should probably create a function to compute that from IOR

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl [hunk 8]
```diff
@@ -71,6 +71,13 @@ TEXTURE2D_ARRAY(_LtcData); // We pack the 3 Ltc data inside a texture array
...
+#define CLEAR_COAT_F0 0.04 // IORToFresnel0(CLEAR_COAT_IOR)
+#define CLEAR_COAT_PERCEPTUAL_ROUGHNESS 0.031622
+#define CLEAR_COAT_ROUGHNESS (CLEAR_COAT_PERCEPTUAL_ROUGHNESS * CLEAR_COAT_PERCEPTUAL_ROUGHNESS) // 0.001
```
Comment (EvgeniiG) [inline ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl]: Best define perceptual roughness and then call a function (inside the macro) to compute the linear roughness.

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl [hunk 9]
```diff
@@ -454,10 +474,7 @@ void EncodeIntoGBuffer( SurfaceData surfaceData,
     }
     else if (surfaceData.materialId == MATERIALID_LIT_CLEAR_COAT)
     {
-        // In the cae of clear coat, we want more precision for the coat normal than for the bottom normal (as it is expected to be smooth). So swap the normal encoding storage in Gbuffer.
-        // It also allow to use clear coat normal for SSR
-        float2 octBottomNormalWS = PackNormalOctEncode(surfaceData.normalWS);
-        outGBuffer2 = float4(octBottomNormalWS * 0.5 + 0.5, surfaceData.coatCoverage, PackFloatInt8bit(surfaceData.coatIOR, (int)(surfaceData.metallic * 15.5f), 16.0) );
+        outGBuffer2 = float4(0.0f, 0.0f, 0.0f, PackFloatInt8bit(surfaceData.coatMask, (int)(surfaceData.metallic * 15.5f), 16.0) );
```
Comment (EvgeniiG) [inline ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl]: Why 15.5?

Comment (EvgeniiG) [reply ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl]: Shouldn't we be using PackInt(), UnpackInt()?

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl [hunk 10]
```diff
@@ -689,62 +701,24 @@ struct PreLightData
...
+    ZERO_INITIALIZE(PreLightData, preLightData);
 
-    float3 N;
-    float  NdotV;
+    float3 N = bsdfData.normalWS;
+    float  NdotV = saturate(dot(N, V));
+    preLightData.clampNdotV = NdotV; // Caution: The handling of edge cases where N is directed away from the screen is handled during Gbuffer/forward pass, so here do nothing
+    preLightData.iblPerceptualRoughness = bsdfData.perceptualRoughness;
 
+    // Clear coat need to modify roughness, V and NdotV for all the other precalculation below
```
Comment (EvgeniiG) [inline ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl]: Misleading comment? You do not modify roughness or NdotV here?

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl [hunk 11]
```diff
@@ -762,7 +736,7 @@ PreLightData GetPreLightData(float3 V, PositionInputs posInput, BSDFData bsdfDat
         // For positive anisotropy values: tangent = highlight stretch (anisotropy) direction, bitangent = grain (brush) direction.
         float3 grainDirWS = (bsdfData.anisotropy >= 0) ? bsdfData.bitangentWS : bsdfData.tangentWS;
         // Reduce stretching for (perceptualRoughness < 0.2).
-        float  stretch = abs(bsdfData.anisotropy) * saturate(5 * bsdfData.perceptualRoughness);
+        float stretch = abs(bsdfData.anisotropy) * saturate(5 * preLightData.iblPerceptualRoughness);
```
Comment (EvgeniiG) [inline ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl]: I don't see why this was changed given that iblPerceptualRoughness = perceptualRoughness

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl [hunk 12]
```diff
@@ -973,47 +935,23 @@ void AccumulateIndirectLighting(IndirectLighting src, inout AggregateLighting ds
 // BSDF share between directional light, punctual light and area light (reference)
 //-----------------------------------------------------------------------------
 
+// This function apply BSDF * cos (required to deal with clear coating)
```
Comment (EvgeniiG) [inline ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl]: What cos? NdotL is applied during light evaluation.

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl [hunk 13]
```diff
@@ -1065,6 +998,25 @@ void BSDF(  float3 V, float3 L, float3 positionWS, PreLightData preLightData, BS
...
+
+        // Add top specular
+        // TODO: Should we call just D_GGX here ?
+        float DV = DV_SmithJointGGX(NdotH, NdotL, NdotV, bsdfData.coatRoughness, preLightData.coatPartLambdaV);
+        specularLighting += coatF * DV;
+
+        // Note: The modification of the base roughness and fresnel0 by the clear coat is already handled in FillMaterialIdClearCoatData
+
+        // Scale diffuse for energy conservation (use NdotL as an approximation)
+        diffuseLighting *= F_Schlick(CLEAR_COAT_F0, NdotL);
```
Comment (EvgeniiG) [inline ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl]: What's happening here? May need more comments...

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl [hunk 14]
```diff
@@ -1186,17 +1140,26 @@ DirectLighting EvaluateBSDF_Punctual(LightLoopContext lightLoopContext,
...
 
-    [branch] if (intensity > 0)
+    // Note: We use NdotL here to early out, but in case of clear coat this is not correct. But we are ok with this
+    [branch] if (intensity > 0.0)
     {
-        // Simulate a sphere light with this hack.
+        // Simulate a sphere light with this hack
+        // Note that it is not correct with our pre-computation of PartLambdaV (mean if we disable the optimization we will not have the
+        // same result) but we don't care as it is a hack anyway
+        if (bsdfData.materialId == MATERIALID_LIT_CLEAR_COAT && HasMaterialFeatureFlag(MATERIALFEATUREFLAGS_LIT_CLEAR_COAT))
```
Comment (EvgeniiG) [inline ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl]: Do you really need a branch here? isn't coatRoughness only used by clear coat?

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl [hunk 15]
```diff
@@ -1297,32 +1257,27 @@ DirectLighting EvaluateBSDF_Line(   LightLoopContext lightLoopContext,
...
+        lighting.diffuse *= (1.0 - preLightData.ltcMagnitudeCoatFresnel);
+        lighting.specular *= (1.0 - preLightData.ltcMagnitudeCoatFresnel);
+        ltcValue = LTCEvaluate(P1, P2, B, preLightData.ltcTransformCoat);
+        lighting.specular += preLightData.ltcMagnitudeCoatFresnel * ltcValue;
     }
 
     // Save ALU by applying 'lightData.color' only once.
-    lighting.diffuse *= lightData.color;
-    lighting.specular *= lightData.color;
+    lighting.diffuse *= lightData.color * lightData.diffuseScale;
```
Comment (EvgeniiG) [inline ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl]: Why move these? Are they now float3s?

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl [hunk 16]
```diff
@@ -1427,37 +1379,29 @@ DirectLighting EvaluateBSDF_Rect(   LightLoopContext lightLoopContext,
...
+        lighting.diffuse *= (1.0 - preLightData.ltcMagnitudeCoatFresnel);
+        lighting.specular *= (1.0 - preLightData.ltcMagnitudeCoatFresnel);
+        ltcValue = PolygonIrradiance(mul(lightVerts, preLightData.ltcTransformCoat));
+        lighting.specular += preLightData.ltcMagnitudeCoatFresnel * ltcValue;
     }
 
     // Save ALU by applying 'lightData.color' only once.
-    lighting.diffuse *= lightData.color;
-    lighting.specular *= lightData.color;
+    lighting.diffuse *= lightData.color * lightData.diffuseScale;
```
Comment (EvgeniiG) [inline ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl]: Same story here.

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl [hunk 17]
```diff
@@ -1580,13 +1525,11 @@ IndirectLighting EvaluateBSDF_Env(  LightLoopContext lightLoopContext,
...
 //    #endif
 
-    weight = 1.0;
-
 #else
 
     // TODO: factor this code in common, so other material authoring don't require to rewrite everything,
     // TODO: test the strech from Tomasz
-    // float roughness       = PerceptualRoughnessToRoughness(bsdfData.perceptualRoughness);
+    // float roughness       = PerceptualRoughnessToRoughness(preLightData.IblPerceptualRoughness);
```
Comment (EvgeniiG) [inline ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl]: Is there any way we can avoid creating yet another roughness variable?
My initial idea was to use roughnessT and roughnessB (both clamped) for direct lighting and perceptualRoughness for IBL.

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl [hunk 18]
```diff
@@ -1724,40 +1669,52 @@ IndirectLighting EvaluateBSDF_Env(  LightLoopContext lightLoopContext,
...
+    }
+    else
+    {
+        // specular transmisted lighting is the remaining of the reflection (let's use this approx)
+        // With refraction, we don't care about the clear coat value, only about the Fresnel, thus why we use 'envLighting ='
+        envLighting = (1.0 - F) * preLD.rgb * preLightData.transmissionTransmittance;
 
-    float4 preLD = SampleEnv(lightLoopContext, lightData.envIndex, R, preLightData.iblMipLevel);
-    envLighting += F * preLD.rgb;
+        // Don't account for clear coat to save ALU
```
Comment (EvgeniiG) [inline ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl]: I don't think it makes sense to have both clear coat and SS refraction, anyway.

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl [hunk 19]
```diff
@@ -1724,40 +1669,52 @@ IndirectLighting EvaluateBSDF_Env(  LightLoopContext lightLoopContext,
...
     UpdateLightingHierarchyWeights(hierarchyWeight, weight);
     envLighting *= weight * lightData.dimmer;
 
     if (GPUImageBasedLightingType == GPUIMAGEBASEDLIGHTINGTYPE_REFLECTION)
-        lighting.specularReflected = envLighting * preLightData.specularFGD;
+        lighting.specularReflected = envLighting;
     else
-        // specular transmisted lighting is the remaining of the reflection (let's use this approx)
-        lighting.specularTransmitted = (1.0 - preLightData.specularFGD) * envLighting * preLightData.transmissionTransmittance;
+        lighting.specularTransmitted = envLighting * preLightData.transmissionTransmittance;
```
Comment (EvgeniiG) [inline ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl]: What is 'transmissionTransmittance'? :-)

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl [hunk 20]
```diff
@@ -1724,40 +1669,52 @@ IndirectLighting EvaluateBSDF_Env(  LightLoopContext lightLoopContext,
...
 
-#endif
+#endif // LIT_DISPLAY_REFERENCE_IBL
 
     UpdateLightingHierarchyWeights(hierarchyWeight, weight);
     envLighting *= weight * lightData.dimmer;
 
     if (GPUImageBasedLightingType == GPUIMAGEBASEDLIGHTINGTYPE_REFLECTION)
-        lighting.specularReflected = envLighting * preLightData.specularFGD;
+        lighting.specularReflected = envLighting;
```
Comment (EvgeniiG) [inline ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl]: Why is FDG gone?

File: ScriptableRenderPipeline/Core/CoreRP/ShaderLibrary/CommonMaterial.hlsl (hunk 4 above)
Comment (sebastienlagarde) [reply ScriptableRenderPipeline/Core/CoreRP/ShaderLibrary/CommonMaterial.hlsl]: done

File: ScriptableRenderPipeline/Core/CoreRP/ShaderLibrary/CommonMaterial.hlsl (hunk 2 above)
Comment (sebastienlagarde) [reply ScriptableRenderPipeline/Core/CoreRP/ShaderLibrary/CommonMaterial.hlsl]: done

File: ScriptableRenderPipeline/Core/CoreRP/ShaderLibrary/CommonMaterial.hlsl (hunk 1 above)
Comment (sebastienlagarde) [reply ScriptableRenderPipeline/Core/CoreRP/ShaderLibrary/CommonMaterial.hlsl]: ConvertF0ForAirInterfaceToF0ForClearCoat15
Done

File: ScriptableRenderPipeline/Core/CoreRP/ShaderLibrary/CommonMaterial.hlsl (hunk 5 above)
Comment (sebastienlagarde) [reply ScriptableRenderPipeline/Core/CoreRP/ShaderLibrary/CommonMaterial.hlsl]: I don't know :)

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/Editor/Material/Lit/LitUI.cs (hunk 6 above)
Comment (sebastienlagarde) [reply ScriptableRenderPipeline/HDRenderPipeline/HDRP/Editor/Material/Lit/LitUI.cs]: done

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl (hunk 8 above)
Comment (sebastienlagarde) [reply ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl]: done

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl (hunk 10 above)
Comment (sebastienlagarde) [reply ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl]: done

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl (hunk 11 above)
Comment (sebastienlagarde) [reply ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl]: due to MipMapLevel processed during cubemap evaluation

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl (hunk 13 above)
Comment (sebastienlagarde) [reply ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl]: done

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl (hunk 14 above)
Comment (sebastienlagarde) [reply ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl]: done

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl (hunk 15 above)
Comment (sebastienlagarde) [reply ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl]: done

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl (hunk 16 above)
Comment (sebastienlagarde) [reply ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl]: done

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl (hunk 17 above)
Comment (sebastienlagarde) [reply ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl]: plenty of usage for roughness (distanceRoughness, attenuation of roughness based on distance from orgin etc...)

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl (hunk 18 above)
Comment (sebastienlagarde) [reply ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl]: done

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl (hunk 20 above)
Comment (sebastienlagarde) [reply ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl]: still here

**Pull request merged**